### DISCUSSION TITLE
Is there a possibility to run a job in argocd after all the apps are deployed and are in sync

### DISCUSSION THREAD

**Discussion by codeShrey** [argoproj/argo-cd#11635] [General]
I want to trigger a workflow from argocd after all the apps are deployed and are in synced status. is there a way to monitor the deployment status of the apps and then trigger the job?

I have tried using post-sync-hooks, but looks like it works only for a single component, suppose I have 3 apps, I dont want to run post-sync-job all the 3 times, after all are deployed I need to trigger once. Please suggest on this.

**Comment by ebuildy** [comment]
Use waves, from https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/ 

Wave will wait for your app become healthy
